### DISCUSSION TITLE
Pytania, problemy

### DISCUSSION THREAD

**Discussion by hawkeye116477** [FiltersHeroes/PolishCookieConsent#76] [Pytania i odp]

**Comment by seba2282** [comment]
@hawkeye116477 Małe pytanie. Czy to wymaga listy I don't care about cookies do poprawnego działania na eu? I czy będzie wersja dla Firefox mobilnego finalnie?

**Reply by admin** [reply]
Jak nie skopiujemy 100% domen z dodatku IDCAC w zasięgu "Europy" to raczej wciąż będzie potrzebny pod "anglojęzyczne" IDCAC.

IDCAC niby ma też skrypty co nie są ograniczone zasięgiem domeny np. `#cadre_alert_cookies` czy nawet OneTrust lub Evidon.

Niby najtrudniejszy będzie np. `chip.de` jak nie ma stabilnych nazw cookies dotyczących pop-upa o RODO/polityce prywatności "ciastek" i ma podobny mankament z odtwarzaniem jak player WP (mrożenie się kodu do końca wykonania jednego).

---

Dopuszczenie do wersji stabilnej Fenix/Firefox Daylight niby wymaga wysłania maila na odpowiedni adres do Moz://i, tak by oni rozpoczęli swoje testy czy nie używamy czasem coś co oni mają na etapie alfa w testowaniu API dodatków.

**Comment by seba2282** [comment]
@krystian3w A czy w planach jest wersja dla mobilnego firefoxa? Jeżeli mogę spytać. Ostatnio I don't care about cookies ma jakieś bugi i nie działa za dobrze, z tego co widzę. Ciasteczkowa zgoda vs dodatek I don't care about cookies - Pierwsze likwiduje irytujące cookie od google, a drugie sobie nie radzi.

**Reply by admin** [reply]
Musiałbyś znaleźć formularz lub ten mail gdzie możemy zgłosić prośbę testów czy dodatek jest zgodny z Firefox Daylight 88+ (hawkeye stwierdza, że był zgodny z Firefox Nightly 89+ na Android).

Ale to może sprawić, że Moz://a będzie sprawdzać tydzień każdą (nawet krytyczną) aktualizację dodatku jakby wdrażać nowości co wydanie a nie hucznie za rok w całości (obecnie jak nie sprawdzają z 100 dodatków przed naszym to w 30 minut dla Firefox jest nowa wersja, w Google okres może wynosić dalej 5 dni, mimo że ostatnio chyba zakończył się do 15 min).

**Comment by seba2282** [comment]
@krystian3w A obejście tego? Zeby jakoś zewnętrznie. Uciążliwe są te ciastka na firefox android.

**Reply by admin** [reply]
Mogę napisać podobną instrukcję jak z `iceraven` (https://github.com/PolishFiltersTeam/PolishCookieConsent#vi-iceraven) dla:

* Fennec z F-droid (https://f-droid.org/en/packages/org.mozilla.fennec_fdroid/) lub wersji Firefox Beta/Nightly

wtedy trzeba by przejść na wersje niestabilną lub spod innych rąk pisaną.

**Comment by seba2282** [comment]
@krystian3w Używam wersji niestabilnej na android aktualnie. Więc przy chwili może spróbuję twojej receptury.

**Comment by admin** [comment]
> Dopuszczenie do wersji stabilnej Fenix/Firefox Daylight niby wymaga wysłania maila na odpowiedni adres do Moz://i, tak by oni rozpoczęli swoje testy czy nie używamy czasem coś co oni mają na etapie alfa w testowaniu API dodatków.

Priorytetowo dodają rozszerzenia z etykietką „Polecane” i w tym przypadku, z tego co doczytałem to pod uwagę biorą też to, czy rozszerzenie przydaje się dla różnych narodowości, więc mogą nam nie przydzielić takiej etykietki, póki nie dołożymy tej zagranicznej bazy ciasteczkowej.

**Comment by seba2282** [comment]
@hawkeye116477 Sami widzicie, Daniel sobie nie radzi za bardzo z EU w swoim dodatku. Tak więc Wasz dodatek teraz działa lepiej. I nikt mi nie powie, że nie mam racji. Przykład podałem z google.

**Reply by admin** [reply]
Przecież mówisz, że to Kiboke musi zaktualizować klikanie pod cookie wall z domeny `consent.`.

A nie działać może z upierdliwości Google do blokad `CSP` (od miesiąca chyba obie domeny w Firefox nie zezwalają na wstrzykiwanie JS - dezaktywowane `security.csp.enable` lub [scriptlet doctor](https://github.com/JustOff/scriptlet-doctor#readme) z dopisanymi domenami `Google.com`/`Google.pl`/`Youtube.com` może stać się wymagane, nie wiem na ile Moz://a lubi grzebanie przy CSP dodatkiem).

Raczej nikt nie będzie się bawić w pisanie `bakeCookie` by subdomena nie zdążyła załadować się.

**Comment by seba2282** [comment]
Czyli winowajcą jest też po części włączone CSP? Czy źle zrozumiałem?

**Reply by admin** [reply]
Może naprawią to za dwa lata na poziomie API dodatków, wtedy taki Scriplet Doctor straci sens.

https://bugzilla.mozilla.org/show_bug.cgi?id=1267027

**Comment by seba2282** [comment]
@krystian3w więc w ubo wystarczy wyłączyć blokowanie csp czy w przeglądarkach to trzeba zdeaktywować o csp?

**Reply by admin** [reply]
Jak nie działa Ci nasze klikanie w obrębie 3 subdomen `consent.*.pl` / `consent.*.com`:

<pre><code> https://<ins><b>consent.</b></ins>google.com/m?continue=
  https://<ins><b>consent.</b></ins>google.pl/m?continue=
https://<ins><b>consent.</b></ins>youtube.com/m?continue=</code></pre>

to zacznij od dodatku https://github.com/JustOff/scriptlet-doctor#readme, dopisz do konfiguracji własnej te 3 subdomeny i sprawdź ponownie.
Jak nie ruszy to wtedy w **about:config** ustaw `security.csp.enable` na **false**. Raczej nie prędko dodatek przejdzie przemianę w aplikacje typu AdGuard (niby wtedy CSP ma niewiele do gadania przy klikaniu).

---

"Egzotycznego" np.: `consent.google.de` 🇩🇪 , `consent.google.com.br` 🇧🇷  nie ma na liście:

https://github.com/PolishFiltersTeam/PolishCookieConsent/blob/84428343eafab0c6b34d9a17141bb2258fb6f227/src/cookieBase/PCB.txt#L58

Jakby miało ruszyć to pewnie jako regex `\.[\w]{2,14}\.?[\w]{2,14}?\/` do czasu implementacji kiedyś zapisu z gwiazdką.

---

Na wersję pop-up jest stosowany html filtering i awaryjne dwa filtry `acis.js`.

**Comment by seba2282** [comment]
@krystian3w Ale napisałem, że tylko Polska Zgoda Ciasteczkowa działa na to. Ale spróbuję dodać te domeny na FF Android, gdzie oficjalnie nie ma jeszcze zgodnego dodatku do ciasteczek.

**Comment by admin** [comment]
@seba2282 
> Tak więc Wasz dodatek teraz działa lepiej

W takim razie dlaczego nie widzę 5-gwiazdkowej recenzji w AMO?

Jak będą pozytywne recenzje, to może prędzej Mozilla doda rozszerzenie do listy na Androida.

**Comment by seba2282** [comment]
@hawkeye116477, zakładam się, że trzeba mieć konto na Mozilla. Szkoda, bo dałbym ocenę.

**Reply by admin** [reply]
> zakładam się, że trzeba mieć konto na Mozilla. Szkoda

Ależ dlaczego szkoda, przecież założenie konta to nic trudnego :stuck_out_tongue:

**Reply by seba2282** [reply]
W sumie racja. Ale następne konto do kolekcji? 😛😛😛

**Reply by admin** [reply]
@seba2282 Coś trzeba kolekcjonować 😛 .

**Reply by seba2282** [reply]
@hawkeye116477 wolę co inne kolekcjonować niż konta. 😛
